apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabTheorycraft/BuildConstaintCard.tsx

```diff
@@ -0,0 +1,152 @@
+import { StatIcon } from '@genshin-optimizer/gi-svgicons'
+import {
+  CustomNumberInput,
+  CustomNumberInputButtonGroupWrapper,
+  DropdownButton,
+} from '@genshin-optimizer/ui-common'
+import { unit } from '@genshin-optimizer/util'
+import ClearIcon from '@mui/icons-material/Clear'
+import {
+  Box,
+  Button,
+  ButtonGroup,
+  CardHeader,
+  Divider,
+  ListItemIcon,
+  ListItemText,
+  MenuItem,
+  Stack,
+} from '@mui/material'
+import { useContext } from 'react'
+import { ArtifactStatWithUnit } from '../../../../Components/Artifact/ArtifactStatKeyDisplay'
+import CardLight from '../../../../Components/Card/CardLight'
+import type { MinTotalStatKey } from '../../../../Database/DataManagers/CharacterTCData'
+import { minTotalStatKeys } from '../../../../Database/DataManagers/CharacterTCData'
+import { CharTCContext } from './CharTCContext'
+
+export function BuildConstaintCard({ disabled }: { disabled: boolean }) {
+  const {
+    charTC: {
+      optimization: { minTotal },
+    },
+    setCharTC,
+  } = useContext(CharTCContext)
+
+  return (
+    <CardLight>
+      <CardHeader title="Stat Constraints" />
+      <Divider />
+      <Box sx={{ p: 1 }}>
+        <Stack spacing={1}>
+          {Object.entries(minTotal).map(([k, v]) => (
+            <Selector key={k} statKey={k} value={v} disabled={disabled} />
+          ))}
+          <DropdownButton disabled={disabled} title={'Add a Stat Constraint'}>
```

Comment:
https://github.com/frzyc/genshin-optimizer/blob/2d83cf45b38d01b25134b10b2694bb05492d81ce/apps/frontend/src/app/Components/StatEditorList.tsx

Please replace with a StatEditorList for ux consistency and so we don't have an extra component


also it'll fix the bug where swapping stats on the constraint doesn't work


Alternatively, using the constraint logic from the optimize tab might be nice. Then users can specify multiple constraints and toggle as desired

Reply:
Both of those UI are very bloated and overt complex for this.

Reply:
I don't think stateditorlist is bloated, it's exactly what this is, with autocomplete. I don't see the need to create another component that does the same thing and also clashes visually/functionally with a component that is already in use

Reply:
i dont want autocomplete, i dont want sorting, i dont want crazy logic for object manipulation that I'll have to patch into `setCharTC`, All i want is a single dropdown and a few inputs.

Reply:
I don't think there's anything I can do to change your mind so ok.

final thoughts:
1. This doesn't align with our separate goal to unify our UI/UX and make things cohesive
2. As much as StatEditorList contains code you might not care about, this is still 50 lines of code that are not strictly necessary to be added/duplicated
3. There is no need to patch setCharTC, you can use it the same way you are currently already using it in the various onChange/onClick methods